app/scripts/directives/overview/serviceInstanceRow.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+angular.module('openshiftConsole').component('serviceInstanceRow', {
+  controller: [
+    '$filter',
+    'DataService',
+    'rowMethods',
+    '$uibModal',
+    ServiceInstanceRow
+  ],
+  controllerAs: 'row',
+  bindings: {
+    apiObject: '<',
+    state: '<'
+  },
+  templateUrl: 'views/overview/_service-instance-row.html'
+});
+
+function ServiceInstanceRow($filter, DataService, rowMethods, $uibModal) {
+  var row = this;
+  _.extend(row, rowMethods.ui);
+
+  // returns the best user-friendly name available for the instance
+  var getDisplayName = function() {
+    var serviceClassName = row.apiObject.spec.serviceClassName;
+    var instanceName = row.apiObject.metadata.name;
+    var serviceClassDisplayName = _.get(row, ['state','serviceClasses', serviceClassName, 'OSBMetadata', 'displayName']);
+    return serviceClassDisplayName || serviceClassName || instanceName;
+  };
+
+  var getBindings = function() {
+    return _.filter(row.state.bindings, function(binding) {
+      return binding.spec.instanceRef.name === row.apiObject.metadata.name;
+    });
+  };
+
+  row.$onChanges = function() {
+    row.notifications = rowMethods.getNotifications(row.apiObject, row.state);
+    row.displayName = getDisplayName();
+    row.instanceBindings = getBindings();
+  };
+
+  row.getSecretForBinding = function(binding) {
+    return binding && row.state.secrets[binding.spec.secretName];
+  }
+
+
+  row.deprovision = function() {
+    // TODO: we probably have to handle bindings in here.
+    // either:
+    // - automatically remove the bindings
+    // - tell the user they must manually unbind before continue
+    $uibModal.open({
+      animation: true,
+      templateUrl: 'views/modals/confirm.html',
+      controller: 'ConfirmModalController',
+      resolve: {
+        modalConfig: function() {
+          return {
+            alerts: {
+              deprovision: {
+                type: 'error',
+                message: 'Service \'' + row.apiObject.spec.serviceClassName + '\' will be deleted and no longer available.'
+              }
+            },
+            detailsMarkup: 'Deprovision Service?',
+            okButtonText: 'Deprovision Service',
```

Comment:
I think this is in the [design docs](https://docs.google.com/document/d/1w2Tor1hgJiNfCClr3q7hBmW6B5zJAqs9saZHn5OJoLY/edit#heading=h.pem26cddew12), but I agree, one word is prob sufficient.